### PR TITLE
[WIP] Add mTLS CA Option on tornjak server

### PR DESCRIPTION
**Pull request by lumjjb**
Addresses https://github.com/spiffe/tornjak/issues/2, by providing a `—mtls-ca` flag to validate mTLS certificates for validation of client certs.

Signed-off-by: Brandon Lum <lumjjb@gmail.com>

### CODE REVIEW

**Comment by mrsabath** [comment]
Checke with @lumjjb if ready to merge

**Comment by mrsabath** [comment]
Testing and documentation (explain the server params, key management)

**Review by mrsabath — APPROVED** [review]
LGTM 

Thank you @lumjjb for this update. This works great! More documentation updates to follow.

**Review by lumjjb — COMMENTED** [review]
Minor nits, else LGTM

File: tornjak-backend/api/manager/server.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ import (
 
 var (
 	jsonContentType string = "application/json"
+	keyShowLen      int    = 40
+	certShowLen     int    = 50
 )
```
Comment (lumjjb) [inline tornjak-backend/api/manager/server.go:21]: use const instead?

File: tornjak-backend/api/manager/server.go [hunk 2]
```diff
@@ -72,8 +74,23 @@ func (s *Server) apiServerProxyFunc(apiPath string) func(w http.ResponseWriter,
 			retError(w, emsg, http.StatusBadRequest)
 			return
 		}
+
+		// Show the end of certs and key for debugging
+		var certSub, caSub, keySub string
+		certStr := string(sinfo.CA)
+		if len(string(certStr)) > certShowLen {
+			certSub = "\n..." + certStr[len(certStr)-certShowLen:len(certStr)]
```
Comment (lumjjb) [inline tornjak-backend/api/manager/server.go:82]: i think you can just do `certStr[len(certStr)-certShowLen:]`

Comment (lumjjb) [reply tornjak-backend/api/manager/server.go:82]: probably better to put all these fields in a slice and then have a loop iterate

File: tornjak-backend/api/agent/server.go [hunk 3]
```diff
@@ -14,6 +14,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/cenkalti/backoff/v3"
```
Comment (lumjjb) [inline tornjak-backend/api/agent/server.go:17]: what's this used for?

Comment (mrsabath) [reply tornjak-backend/api/agent/server.go]: removed...

**Pull request merged**